daemon/control.c

```diff
@@ -337,17 +344,22 @@ control_on_users_changed(control_t *self)
                    (unsigned)uid,
                    users_user_exists(users, uid) ? "exists" : "n/a");
     }
+
+    later_schedule(self->ctl_rethink_settings);
+    // -> control_rethink_settings_cb()
 }
 
 void
-control_on_session_changed(control_t *self)
+control_on_session_changed(control_t *self, uid_t prev, uid_t next)
```

Comment:
In general I've tried to avoid interfaces where something "dictates" what the previous state was (i.e. if previous state matters to receiver, then the receiver should keep track of what it used to be). But that is just personal preference - this works too and keeps things simple for the purpose at hand. So, no need to make any changes.

Reply:
Changed this too. I agree that it's better like this.